### PR TITLE
adding a new view to query the sync logs by date

### PR DESCRIPTION
**Pull request by calellowitz**
@proteusvacuum cc: @snopoke fyi: @ctsims 
This adds a view to query sync logs by date. Any time there is a bug related to syncing going forward we will be able to quickly identify all affected users. It would have been useful in determining the scope of both the opened_date and multimedia bugs, and could help in recovery efforts for future sync related bugs. Not sure how long the couch preindex of this will take so i think its probably worth holding off the merge until https://github.com/dimagi/commcare-hq/pull/12556 is deployed

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/_design/views/sync_logs_by_date/map.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function(doc) {
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/_design/views/sync_logs_by_date/map.js]: Parsing error: Unexpected token (

**Comment by calellowitz** [comment]
✅

**Comment by proteusvacuum** [comment]
Do you think we should wait until tomorrow to merge this? (for the reindex?)

**Comment by calellowitz** [comment]
i was hoping to merge it now, then kick off a preindex assuming it would be done by normal deploy time so that tomorrow we could run the data that @ctsims wanted. but if you think waiting until friday is safer, im ok holding off.

**Comment by ctsims** [comment]
@calellowitz I don't think there's a specific rush to get it out at this point, more will be useful to ensure we can do at some point to compare and understand the impact.

**Comment by czue** [comment]
sync logs are in their own DB so i expect this will be pretty fast. also it would be fine to deploy before preindex completes since nothing in the codebase actually referecnes this.

**Comment by proteusvacuum** [comment]
@calellowitz merged

**Comment by esoergel** [comment]
sync logs are in both their own db *and* the main `commcarehq` db, as are any views, meaning that this has to operate on the main db too.  ( @calellowitz and I just looked in to why the preindex is taking forever)

**Comment by czue** [comment]
dang, sorry about the bad information. 

also i overlooked that this was adding a view to the `phone` design doc. I think we are trying to always make new views their own design docs moving forwards which also would also help avoid the massive preindex. is there any chance we want to revert this and roll it out in an independent design/view using `couchapps`?

**Comment by calellowitz** [comment]
@czue @esoergel will https://github.com/dimagi/commcare-hq/issues/12579 be sufficient or will i need to do something else to stop/avoid the ongoing preindex

**Comment by czue** [comment]
> @czue @esoergel will #12579 be sufficient or will i need to do something else to stop/avoid the ongoing preindex

I think in the past we have had to:

1. delete the -tmp design doc
2. (sometimes) tell cloudant so that they can kill the indexing process

Though I think what probably happened here is that the next deploy overwrote the `phone-tmp` design doc to be the same as the `phone` one, which hopefully was the equivalent of deleting it. I don't know how to check whether the indexing process is still running, but I'm guessing it's sorted itself out by this point. So the shorter answer to your question is likely no additional action needed.

fyi @dannyroberts is the resident expert on this stuff in case you have questions in the future.

**Comment by calellowitz** [comment]
@czue For future reference if this happens again, it was enough to just deploy that fix. The new preindex did overwrite the `phone-tmp` design doc, which apparently was enough to kill the existing preindex. The cloudant dashboard has a list of active tasks in the cluster and you can track any indexing tasks there, and   as soon as the new deploy command was run, there were no more active tasks on the main db and a fresh one on the synclog db.

**Comment by dannyroberts** [comment]
Yeah that behavior makes sense to me. The thing about manually having to kill a reindex may have only happened as a result of an edge case that used to happen where we'd attempt to delete an index, and instead one of its ghost conflict documents popped up to take its place. Overwriting like we did this time may trigger a different sequence of events that includes killing the old processes; or maybe we just got lucky.

**Pull request merged**